src/main/java/com/mazyavr/schedule/controller/UserController.java

```diff
@@ -57,68 +61,101 @@ public class UserController {
   @Autowired
   private TaskRepository taskRepository;
 
-  private String token;
+  private String refreshToken;
+  private String googleId;
   private Integer lifespan;
 
   private Calendar initCalendar(long userId) {
 
     Calendar userCalendar = null;
 
-    String userToken = userRepository.findById(userId).get().getToken();
+    String userToken = userRepository.findById(userId).get().getRefreshToken();
 
     if (userToken == null) {
       throw new RuntimeException("Unable to initialize google calendar");
     }
-
+  
     try {
       var HTTP_TRANSPORT = GoogleNetHttpTransport.newTrustedTransport();
       var credential = new GoogleCredential().setAccessToken(userToken);
-
-      userCalendar = new Calendar.Builder(HTTP_TRANSPORT, JSON_FACTORY, credential)
-          .setApplicationName("DownScheduler")
-          .build();
+      
+      return new Calendar.Builder(HTTP_TRANSPORT, JSON_FACTORY, credential)
+        .setApplicationName("DownScheduler")
+        .build();
     } catch (IOException | GeneralSecurityException e) {
-
-    }
-
-    if (userCalendar == null) {
       throw new RuntimeException("Unable to initialize google calendar");
     }
-    return userCalendar;
   }
-
-  /**
-   * Здесь реализована только регистрация(добавление в БД), но не авторизация(обновление юзверя в
-   * БД)
-   */
+  
   @PostMapping(path = "/login")
   public @ResponseBody RedirectView login(
       @RequestParam(value = "code") String code,
       @RequestParam(value = "state") String redirectUri,
       HttpServletResponse servletResponse
   ) {
     try {
+      var HTTP_TRANSPORT = new NetHttpTransport();
+      var CLIENT_ID = "530835646351-odkl0qt3l07n6s7httrpsd6rd37ka5l9.apps.googleusercontent.com";
+      
       GoogleTokenResponse response = new GoogleAuthorizationCodeTokenRequest(
-          new NetHttpTransport(), new GsonFactory(),
-          "530835646351-odkl0qt3l07n6s7httrpsd6rd37ka5l9.apps.googleusercontent.com",
-          "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",
-          code, "http://localhost:8080/users/login")
-          .execute();
-      this.token = response.getAccessToken();
+        HTTP_TRANSPORT, new GsonFactory(), CLIENT_ID,
+        "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",
+        code, "http://localhost:8080/users/login")
+        .execute();
+  
+      this.refreshToken = response.getAccessToken();
+      var idToken = response.getIdToken();
+      var verifier = new GoogleIdTokenVerifier.Builder(HTTP_TRANSPORT, JSON_FACTORY)
+        .setAudience(Collections.singletonList(CLIENT_ID)).build();
+  
+      var verifiedToken = verifier.verify(idToken);
+      
+      if (verifiedToken != null) {
+        googleId = verifiedToken.getPayload().getSubject();
+      }
```

Comment:
Если `verifiedToken == null`, то мы н можем проверить зарегистрирован ли пользователь. Можно кидать исключение, если проверка не прошла. И ещё, переменная `googleId` должна быть локальная, а не поле класса, её значение вне этого метода не должно быть доступно

```suggestion
      if (verifiedToken == null) {
        throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "No user ID provided");
      }
      
      var googleId = verifiedToken.getPayload().getSubject();
```